src/main/java/net/shadowmage/ancientwarfare/vehicle/config/AWVehicleStatics.java

```diff
@@ -1,45 +1,281 @@
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 import net.shadowmage.ancientwarfare.core.config.ModConfiguration;
+import net.shadowmage.ancientwarfare.vehicle.AncientWarfareVehicles;
 
+@Config(modid = AncientWarfareVehicles.MOD_ID, name = "ancientwarfare\\AncientWarfareVehicle")
 public class AWVehicleStatics extends ModConfiguration {
 
+	@SuppressWarnings("unused")
+	@Mod.EventBusSubscriber(modid = AncientWarfareVehicles.MOD_ID)
+	private static class EventHandler {
+		/**
+		 * Inject the new values and save to the config file when the config has been changed from the GUI.
+		 *
+		 * @param event The event
+		 */
+		@SubscribeEvent
+		public static void onConfigChanged(ConfigChangedEvent.OnConfigChangedEvent event) {
+			if (event.getModID().equals(AncientWarfareVehicles.MOD_ID)) {
+				ConfigManager.sync(AncientWarfareVehicles.MOD_ID, Config.Type.INSTANCE);
+			}
+		}
+	}
+
+	@Config.Ignore
 	public static final String KEY_VEHICLE_FORWARD = "keybind.vehicle.forward";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_REVERSE = "keybind.vehicle.reverse";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_LEFT = "keybind.vehicle.left";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_RIGHT = "keybind.vehicle.right";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_FIRE = "keybind.vehicle.fire";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_ASCEND_AIM_UP = "keybind.vehicle.ascend.aim.up";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_DESCEND_AIM_DOWN = "keybind.vehicle.descend.aim.down";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_AMMO_PREV = "keybind.vehicle.ammo.prev";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_AMMO_NEXT = "keybind.vehicle.ammo.next";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_TURRET_LEFT = "keybind.vehicle.turret.left";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_TURRET_RIGHT = "keybind.vehicle.turret.right";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_MOUSE_AIM = "keybind.vehicle.mouse.aim";
+	@Config.Ignore
 	public static final String KEY_VEHICLE_AMMO_SELECT = "keybind.vehicle.ammo.select";
 
+	@Config.Ignore
+	public static final String ClientAndServerSideNote = "Affect both client and server. These configs must match for client and server, or strange and probably BAD things WILL happen.";
+
 	public AWVehicleStatics(String mod) {
 		super(mod);
 	}
 
-	//TODO add configuration for these values (probably annotations) - descriptions in 1.6.4 AW
-	public static boolean oversizeAmmoEnabled = true;
+	@Config.Name("General Settings")
+	@Config.Comment(ClientAndServerSideNote)
+	public static GeneralSettings generalSettings = new GeneralSettings();
+
+	public static class GeneralSettings {
+
+		// unused
+		//	@Config.Name("fire_block_break_events")
+		//	@Config.Comment("")
+		//	@Config.Ignore // unused
+		//	public boolean fireBlockBreakEvents = true;
```

Comment:
why is this commented out code here? if we don't need it, let's just remove. we can add it in the future easily if needed.